### PR TITLE
Feature/add notification from core

### PR DESCRIPTION
**Pull request by davidcassell**
The PR contains the changes to move Core.Notifications to Easify.

New Features

- #72 Add support for Core.Notification to the framework

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/Easify.Extensions.Notifications.Messaging.MailKit/MailKitMessagingService.cs [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+                await client.SendAsync(message);
+                await client.DisconnectAsync(true);
+            }
+        }
+
+        private MailboxAddress CreateAddress(EmailAddress recipient)
+        {
+            if (recipient == null) throw new ArgumentNullException(nameof(recipient));
+
+            return new MailboxAddress(recipient.Name ?? recipient.Email, recipient.Email);
```
Comment (LewisPercival) [inline src/Easify.Extensions.Notifications.Messaging.MailKit/MailKitMessagingService.cs:82]: Is it right to use Email for the name when name is blank?

File: README.md [hunk 2]
```diff
@@ -1,32 +1,82 @@
...
+    "LocalDomain": "ICGPLC",
+    "Port": "Optional. Default to 25"
+}
+
+```
+
+Then you can inject **INotificationService** to your classes to be able to send the email.
+
+**Note:** In ICG we are using Relay Messaging so there is no setup to be made on application servers to support smtp. Also the sender can be a fake email address or even a descriptive email address like
+*mezzanine-autosender@icgam.com*
```
Comment (LewisPercival) [inline README.md]: Should this be there for opensource?

File: src/Easify.Extensions.Notifications.Messaging/EmailAddressExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+// 
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace Easify.Extensions.Notifications.Messaging
+{
+    public static class EmailAddressExtensions
+    {
+        public const string AddressSeparator = ";";
+        public const string EmailAndNameSeparator = "|";
```
Comment (LewisPercival) [inline src/Easify.Extensions.Notifications.Messaging/EmailAddressExtensions.cs:26]: Should this be | or . ?

File: src/Easify.Extensions.Notifications.Templating.HandleBars/HandleBarsTemplateContentRenderer.cs [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+using System.Threading.Tasks;
+using FuManchu;
+
+namespace Easify.Extensions.Notifications.Templating.HandleBars
+{
+    public sealed class HandleBarsTemplateContentRenderer : ITemplateContentRenderer
+    {
+        public HandleBarsTemplateContentRenderer()
+        {
+            Handlebars.RegisterHelper("date", options => $"{options.Data:dd/MM/yyyy}");
```
Comment (LewisPercival) [inline src/Easify.Extensions.Notifications.Templating.HandleBars/HandleBarsTemplateContentRenderer.cs:27]: Is this date format valid? should be configurable?

**Review by LewisPercival — COMMENTED** [review]
Some comments to look at

File: src/Easify.Extensions.Notifications/NotificationService.cs [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+namespace Easify.Extensions.Notifications
+{
+    public class NotificationService : INotificationService
+    {
+        private readonly IMessagingService _messagingService;
+        private readonly NotificationOptions _options;
+        private readonly ITemplateRenderer _templateRenderer;
+        private readonly ILogger<NotificationService> _logger;
+
+        // TODO: Think on moving this to a dependent service
```
Comment (french-justin) [inline src/Easify.Extensions.Notifications/NotificationService.cs:37]: Is this for future or needs to be done now?

**Review by french-justin — APPROVED** [review]
This looks good. Got one comment around a todo and you should remove the ICG example as per Lewis' comment.
Happy to approve though

File: src/Easify.Extensions.Notifications.Messaging.MailKit/MailKitMessagingService.cs (hunk 1 above)
Comment (davidcassell) [reply src/Easify.Extensions.Notifications.Messaging.MailKit/MailKitMessagingService.cs:82]: This is the same as in ICG.Core.Notification. It's lift and shift.

File: README.md (hunk 2 above)
Comment (davidcassell) [reply README.md]: This has been changed

File: src/Easify.Extensions.Notifications.Templating.HandleBars/HandleBarsTemplateContentRenderer.cs (hunk 4 above)
Comment (davidcassell) [reply src/Easify.Extensions.Notifications.Templating.HandleBars/HandleBarsTemplateContentRenderer.cs:27]: This is the same as in ICG.Core.Notification. It's lift and shift.

File: src/Easify.Extensions.Notifications/NotificationService.cs (hunk 5 above)
Comment (davidcassell) [reply src/Easify.Extensions.Notifications/NotificationService.cs:37]: Let's leave this for now. It's a good one as we shouldn't be new-ing this up. But it's FluentValidation and so static at this time.

File: src/Easify.Extensions.Notifications.Messaging/EmailAddressExtensions.cs (hunk 3 above)
Comment (davidcassell) [reply src/Easify.Extensions.Notifications.Messaging/EmailAddressExtensions.cs:26]: This is the same as in ICG.Core.Notification. It's lift and shift.

**Pull request merged**